src/metabase/api/card.clj

```diff
@@ -735,26 +734,28 @@
   (when new-collection-id-or-nil
     (events/publish-event! :event/collection-touch {:collection-id new-collection-id-or-nil :user-id api/*current-user-id*})))
 
-#_{:clj-kondo/ignore [:deprecated-var]}
-(api/defendpoint POST "/collections"
+(api.macros/defendpoint :post "/collections"
   "Bulk update endpoint for Card Collections. Move a set of `Cards` with `card_ids` into a `Collection` with
   `collection_id`, or remove them from any Collections by passing a `null` `collection_id`."
-  [:as {{:keys [card_ids collection_id]} :body}]
-  {card_ids      [:sequential ms/PositiveInt]
-   collection_id [:maybe ms/PositiveInt]}
+  [_route-params
+   _query-params
+   {:keys [card_ids collection_id]} :- [:map
+                                        [:card_ids      [:sequential ms/PositiveInt]]
+                                        [:collection_id {:optional true} [:maybe ms/PositiveInt]]]]
   (move-cards-to-collection! collection_id card_ids)
   {:status :ok})
 
 ;;; ------------------------------------------------ Running a Query -------------------------------------------------
 
-#_{:clj-kondo/ignore [:deprecated-var]}
-(api/defendpoint POST "/:card-id/query"
+(api.macros/defendpoint :post "/:card-id/query"
   "Run the query associated with a Card."
-  [card-id :as {{:keys [parameters ignore_cache dashboard_id collection_preview], :or {ignore_cache false dashboard_id nil}} :body}]
-  {card-id            ms/PositiveInt
-   ignore_cache       [:maybe :boolean]
-   collection_preview [:maybe :boolean]
-   dashboard_id       [:maybe ms/PositiveInt]}
+  [{:keys [card-id]} :- [:map
+                         [:card-id ms/PositiveInt]]
+   _query-params
+   {:keys [parameters ignore_cache dashboard_id collection_preview], :or {ignore_cache false dashboard_id nil}} :- [:map
+                                                                                                                    [:ignore_cache       {:optional true} [:maybe :boolean]]
+                                                                                                                    [:collection_preview {:optional true} [:maybe :boolean]]
+                                                                                                                    [:dashboard_id       {:optional true} [:maybe ms/PositiveInt]]]]
```

Comment:
```suggestion
   {:keys [parameters ignore_cache dashboard_id collection_preview],
    :or {ignore_cache false dashboard_id nil}}
   :- [:map
       [:ignore_cache       {:optional true} [:maybe :boolean]]
       [:collection_preview {:optional true} [:maybe :boolean]]
       [:dashboard_id       {:optional true} [:maybe ms/PositiveInt]]]]
```

Reply:
Moved to https://github.com/metabase/metabase/pull/52569